Assignment 2

iwissemben opened the pull request:
Last commit for this branch:

- The aim was to show if there is a difference in PSD estimation across 3 methods (FFT,periodogram(),welch) and between matlab and python.

- In comparisons_1channel1000Hz.ipynb you can see my results, i computed the RMSE for each method to compare matlab and python results. I also computed the distance between the results (absolute difference) among frequencies which i plotted in graphs.

- I also  tweaked the parameters of the scipy functions, to get results as close as possible to matlabs. Finally, i tried to explain why a difference is there by reading matlab and scipy's documentation to understand the nuances between the functions. 

All this led me to discover aspects that i haven't noticied in my initial use of scipy.signal.welch() used to computing the ERSP.

Please review the results before I send them to Pr.Ramdani

iwissemben commented:
Bonjour @DenisMot  j'ai eu le retour du professeur Ramdani sur la comparaison des fonctions matlab et python. Je pense que l'on peut merge. Pourrais-tu approuver? Dans la foulée je soumettrais une aure pull request pour les signaux artificiels et la nouvelle methode de calcul des ERSP sur signaux artificiels.

DenisMot commented:
I suggest a few changes to have all pieces of code running on both OSX and W11 and to have the jupyter files more clearly describe the problems at hand : 

- [ ]  add a markdown + code bloc in all jupyter files  

``` markdown 
## Ensure that the root directory is the project directory
In some cases, the root directory may not be the project directory. This is usually the case when the notebook is run from a subdirectory of the project directory.
```

``` python 
# if pwd is SRC  change to root
if os.getcwd().split("/")[-1] == "SRC":
    os.chdir("..")
print("Current working directory: ", os.getcwd())
```

- [ ]  Remove the dependency to `%matplotlib widget`  in the jupyter files  
    1. the widget is handy for exploration, but detrimental to reproducibility and explications. 
    1. for example in `comparisons_1channel1000Hz.ipynb`, an in depth analysis of the problem should result in highlights on *where things are important*. Hence, I would expect that you :  
        -  provide the global plot (already done)
        - add a second plot where you **chose** the right scale to make the visual comparison *where it matters*. Indeed, the choices necessary to decide  *where it matters* are not trivial, they are part of the scientific analysis of the problem. It is likely that this second plot deserves a markdown box to explain the choices and describe (textually) the highlights.  It is also likely that the plot will superimpose the curves of interest to better visualize the differences. 
        - alternatively, from a technical standpoint and a different visualisation strategy, you might get inspiration from https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib)](https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib.

iwissemben commented:
Hi, thank you for this review:

> I suggest a few changes to have all pieces of code running on both OSX and W11 and to have the jupyter files more clearly describe the problems at hand :
> 
> * [x]   add a markdown + code bloc in all jupyter files
> 
> ```gfm
> ## Ensure that the root directory is the project directory
> In some cases, the root directory may not be the project directory. This is usually the case when the notebook is run from a subdirectory of the project directory.
> ```
> 
> ```python
> # if pwd is SRC  change to root
> if os.getcwd().split("/")[-1] == "SRC":
>     os.chdir("..")
> print("Current working directory: ", os.getcwd())
> ```
> 
- Thanks for the suggestion, I forgot that using the project directory as root wasn't default. This settings customization (i made on my VSCode) may indeed lead to issues when using jupyter or another machine with different setup.  
-  I added the bit of code to all scripts (.py and .ipynb) that require to navigate in the directory.
> * [x]   Remove the dependency to `%matplotlib widget`  in the jupyter files
>   
>   1. the widget is handy for exploration, but detrimental to reproducibility and explications.
>   2. for example in `comparisons_1channel1000Hz.ipynb`, an in depth analysis of the problem should result in highlights on _where things are important_. Hence, I would expect that you :
>      
>      * provide the global plot (already done)
>      * add a second plot where you **chose** the right scale to make the visual comparison _where it matters_. Indeed, the choices necessary to decide  _where it matters_ are not trivial, they are part of the scientific analysis of the problem. It is likely that this second plot deserves a markdown box to explain the choices and describe (textually) the highlights.  It is also likely that the plot will superimpose the curves of interest to better visualize the differences.
>      * alternatively, from a technical standpoint and a different visualisation strategy, you might get inspiration from [https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib)](https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib](https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib)%5D(https://stackoverflow.com/questions/13583153/how-to-zoomed-a-portion-of-image-and-insert-in-the-same-plot-in-matplotlib).

- Yes! We've seen that the widget does not behave the same on windows and macOS. I removed it.
- As you suggested instead, I superimposed the Matlab PSD results to Python's on the subplots and also added inset_axes to the subplots. This will help the reader better see the difference in the distributions.
